### PR TITLE
Manually retry failing task upsert queries

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #4428
https://pulp.plan.io/issues/4428

### CODE REVIEW

File: server/pulp/server/async/tasks.py [hunk 1]
```diff
@@ -514,11 +515,20 @@ def __call__(self, *args, **kwargs):
             worker_name = self.request.hostname
             # Using 'upsert' to avoid a possible race condition described in the apply_async method
             # above.
-            TaskStatus.objects(task_id=self.request.id).update_one(
-                set__state=constants.CALL_RUNNING_STATE,
-                set__start_time=start_time,
-                set__worker_name=worker_name,
-                upsert=True)
+            try:
```
Comment (dkliban) [inline server/pulp/server/async/tasks.py:518]: This is where the sleep statement would have helped.

**Review by dkliban — APPROVED** [review]
These changes make sense to me. I was misunderstanding the problem before. I commented on the line where we could try a sleep statement. However, I think these changes should suffice.

File: server/pulp/server/async/tasks.py [hunk 2]
```diff
@@ -514,11 +515,20 @@ def __call__(self, *args, **kwargs):
...
+                TaskStatus.objects(task_id=self.request.id).update_one(
+                    set__state=constants.CALL_RUNNING_STATE,
+                    set__start_time=start_time,
+                    set__worker_name=worker_name,
+                    upsert=True)
+            except DuplicateKeyError:
```
Comment (daviddavis) [inline server/pulp/server/async/tasks.py]: This should be `NotUniqueError`

**Comment by evgeni** [comment]
This patch fixed the E11000 errors I've been seeing when syncing multiple repos in bulk to a 2.18 install.

**Review by dkliban — APPROVED** [review]
Thank you!

**Pull request merged**